### DISCUSSION TITLE
Please describe the steps to be taken while running the Request-Reply communication in fastdds

### DISCUSSION THREAD

**Discussion by Akul1903** [eProsima/Fast-RTPS-docs#903] [General]
As here described in the link: https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/request_reply/request_reply.html#request-reply-communication

Regarding the working of request reply communication, But still not able to recognize like with the help of which script we need to genrate the executable files (for example when working in publisher subscriber model we use fastddsgen to genrate files).

and after genrating files how to execute client and server for request reply kindly help me in that

**Comment by JesusPoderoso** [comment]
Hi @Akul1903, the Fast DDS [request-reply example](https://github.com/eProsima/Fast-DDS/tree/master/examples/cpp/request_reply) may include some hints for you.

**Reply by Akul1903** [reply]
Hi @JesusPoderoso Sorry for disturbing you again

But this repository also does't contain enough information how to get output and other such doubts

**Reply by JesusPoderoso** [reply]
Using an equivalent IDL, you can generate the types using Fast DDS-Gen.
The implementation you would need to perform to deploy an RPC communication is part of the example server and client code.

**Reply by Akul1903** [reply]
do i need to run the output using the same publisher subscriber type of terminal
with ./target publisher or ./target subscriber

**Reply by JesusPoderoso** [reply]
In each example, there is a [README](https://github.com/eProsima/Fast-DDS/blob/master/examples/cpp/request_reply/README.md) file with information you should read about [the content](https://github.com/eProsima/Fast-DDS/tree/master/examples/cpp/request_reply#description-of-the-example) and [how to run the example](https://github.com/eProsima/Fast-DDS/tree/master/examples/cpp/request_reply#run-the-example).
